src/ciphers/des.c

```diff
@@ -1963,28 +1929,31 @@ int des_test(void)
         ***/
     };
     int i, y;
-    unsigned char tmp[8];
+    unsigned char ct[8], tmp[8];
     symmetric_key des;
 
     for(i=0; i < (int)(sizeof(cases)/sizeof(cases[0])); i++)
     {
         if ((err = des_setup(cases[i].key, 8, 0, &des)) != CRYPT_OK) {
-           return err;
+            return err;
         }
-        if (cases[i].mode != 0) {
-           des_ecb_encrypt(cases[i].txt, tmp, &des);
-        } else {
-           des_ecb_decrypt(cases[i].txt, tmp, &des);
+
+        des_ecb_encrypt(cases[i].txt, ct, &des);
+
+        if (compare_testvector(ct, sizeof(ct), cases[i].out, 8, "DES Encrypt", i) != 0) {
+            return CRYPT_FAIL_TESTVECTOR;
         }
 
-        if (compare_testvector(cases[i].out, sizeof(tmp), tmp, sizeof(tmp), "DES", i) != 0) {
-           return CRYPT_FAIL_TESTVECTOR;
+        des_ecb_decrypt(ct, tmp, &des);
+
+        if (compare_testvector(tmp, sizeof(tmp), cases[i].txt, 8, "DES Decrypt", i) != 0) {
+            return CRYPT_FAIL_TESTVECTOR;
         }
 
-        /* now see if we can encrypt all zero bytes 1000 times, decrypt and come back where we started */
+        /* now see if we can encrypt all zero bytes 100 times, decrypt and come back where we started */
```

Comment:
Wouldn't it be enough if we do this with one key with 1000 rounds?

Now I also realize why I excluded those additional testvectors, I only remember that it took forever to run when I enabled them, but that makes sense now :-)

Reply:
Each iteration only encrypts and decrypts 8 bytes, so there's not a huge amount of actual work being done even with 1000 iterations for every test vector. Running the 1000 iterations test outside the test vector loop does seem cleaner and more logical though, so probably worth doing for that reason.